ecosystem/indexer-grpc/indexer-grpc-fullnode/src/stream_coordinator.rs

```diff
@@ -80,69 +83,76 @@ impl IndexerStreamCoordinator {
+        let mut task_batches = vec![];
+        let mut current_batch = vec![];
+        let mut current_batch_size = 0;
+        for (txn, size) in sorted_transactions_from_storage_with_size {
+            current_batch.push(txn);
+            current_batch_size += size;
+            if current_batch_size > MINIMUM_TASK_LOAD_SIZE_IN_BYTES {
+                task_batches.push(current_batch);
+                current_batch = vec![];
+                current_batch_size = 0;
+            }
+        }
+        if !current_batch.is_empty() {
+            task_batches.push(current_batch);
+        }
+
+        let output_batch_size = self.output_batch_size;
         let ledger_chain_id = self.context.chain_id().id();
         let mut tasks = vec![];
-        let batches = self.get_batches().await;
-        let output_batch_size = self.output_batch_size;
-
-        for batch in batches {
+        for batch in task_batches {
             let context = self.context.clone();
-            let ledger_version = self.highest_known_version;
-            let transaction_sender = self.transactions_sender.clone();
-
             let task = tokio::spawn(async move {
-                let fetch_start_time = std::time::Instant::now();
-                // Fetch and convert transactions from API
-                let raw_txns =
-                    Self::fetch_raw_txns_with_retries(context.clone(), ledger_version, batch).await;
-                let first_raw_transaction = raw_txns.first().unwrap();
-                let last_raw_transaction = raw_txns.last().unwrap();
-                let mut last_transaction_timestamp = None;
-                if enable_expensive_logging {
-                    // Reusing the conversion methods which need a vec, so make a vec of size 1
-                    let api_txn = Self::convert_to_api_txns(context.clone(), vec![
-                        last_raw_transaction.clone(),
-                    ])
-                    .await;
-                    let pb_txn = Self::convert_to_pb_txns(api_txn);
-                    last_transaction_timestamp = pb_txn.first().unwrap().timestamp.clone();
-                }
-                log_grpc_step_fullnode(
-                    IndexerGrpcStep::FullnodeFetchedBatch,
-                    Some(first_raw_transaction.version as i64),
-                    Some(last_raw_transaction.version as i64),
-                    last_transaction_timestamp.as_ref(),
-                    Some(ledger_version as i64),
-                    None,
-                    Some(fetch_start_time.elapsed().as_secs_f64()),
-                    Some(raw_txns.len() as i64),
-                );
-
-                let convert_start_time = std::time::Instant::now();
-                let api_txns = Self::convert_to_api_txns(context, raw_txns).await;
+                let raw_txns = batch;
+                let api_txns = Self::convert_to_api_txns(context, raw_txns);
                 api_txns.last().map(record_fetched_transaction_latency);
```

Comment:
we don't need this metric anymore since we already record transaction timestamp everywhere else

Reply:
done.